#
# Committed-By: Marc Kleine-Budde <mkl@pengutronix.de>, 2006-11-29
#
# Error:
#
# inclusion of hosts dirs is not desired if cross compiling
#
# State:
#
# should be commited
#
---
# configure.ac |   16 +++++++++++-----
# 1 file changed, 11 insertions(+), 5 deletions(-)
#
Index: pure-ftpd-1.0.28/configure.ac
===================================================================
--- pure-ftpd-1.0.28.orig/configure.ac
+++ pure-ftpd-1.0.28/configure.ac
@@ -29,6 +29,7 @@ AC_ARG_VAR(PYTHON,local path to the pyth
 python_possible_path="/usr/bin:/usr/local/bin:/bin:/opt/python/bin:/opt/python/usr/bin:/opt/python/usr/local/bin"
 AC_PATH_PROG(PYTHON,python,/usr/bin/env python,$python_possible_path)
 
+if test "$cross_compiling" != yes; then
 if test -d /usr/local/include; then
   CPPFLAGS="$CPPFLAGS -I/usr/local/include"
 fi
@@ -44,6 +45,7 @@ fi
 if uname | fgrep SunOS > /dev/null 2> /dev/null ; then
   CPPFLAGS="$CPPFLAGS -D_XPG4_2=1"
 fi
+fi
 
 dnl Checks for header files
 
@@ -1196,8 +1198,10 @@ AC_ARG_WITH(mysql,
       LDFLAGS="$LDFLAGS -L${withval}/lib -L${withval}/lib/mysql -L${withval}/mysql/lib"
       CPPFLAGS="$CPPFLAGS -I${withval}/include -I${withval}/include/mysql -I${withval}/mysql/include"
     else
-      CFLAGS="$CFLAGS `mysql_config --cflags`"
-      LDFLAGS="$LDFLAGS `mysql_config --libs`"
+      if test "$cross_compiling" != yes; then
+        CFLAGS="$CFLAGS `mysql_config --cflags`"
+        LDFLAGS="$LDFLAGS `mysql_config --libs`"
+      fi
     fi
     AC_CHECK_LIB(m, floor)
     AC_CHECK_LIB(z, gzclose)
@@ -1230,9 +1234,11 @@ AC_ARG_WITH(pgsql,
       LDFLAGS="$LDFLAGS -L${withval}/lib -L${withval}/lib/pgsql -L${withval}/lib/postgresql -L${withval}/pgsql/lib -L${withval}/postgresql/lib"
       CPPFLAGS="$CPPFLAGS -I${withval}/include -I${withval}/include/pgsql -I${withval}/include/postgresql -I${withval}/pgsql/include -I${withval}/postgresql/include"
     else
-      LD_RUN_PATH="`pg_config --libdir`${LD_RUN_PATH:+:}${LD_RUN_PATH}"
-      CPPFLAGS="$CPPFLAGS -I`pg_config --includedir`"
-      LDFLAGS="$LDFLAGS -L`pg_config --libdir`"
+      if test "$cross_compiling" != yes; then
+        LD_RUN_PATH="`pg_config --libdir`${LD_RUN_PATH:+:}${LD_RUN_PATH}"
+        CPPFLAGS="$CPPFLAGS -I`pg_config --includedir`"
+        LDFLAGS="$LDFLAGS -L`pg_config --libdir`"
+      fi
     fi
     AC_CHECK_LIB(m, floor)
     AC_CHECK_LIB(z, gzclose)
